inference-engine/src/gna_plugin/gna_device.cpp

```diff
@@ -40,15 +43,36 @@ uint8_t* GNADeviceHelper::alloc(uint32_t size_requested, uint32_t *size_granted)
 #else
     const auto status = Gna2MemoryAlloc(size_requested, size_granted, &memPtr);
     checkGna2Status(status, "Gna2MemoryAlloc");
+    gnalog() << "Gna2MemoryAlloc(" << size_requested << ") -> " << *size_granted << ", " << memPtr << "\n";
 #endif
     if (memPtr == nullptr) {
         THROW_GNA_EXCEPTION << "GNAAlloc failed to allocate memory. Requested: " << size_requested << " Granted: " << *(size_granted);
     }
+
     dumpXNNROPtr = memPtr;
     dumpXNNROSize = *size_granted;
     return static_cast<uint8_t *>(memPtr);
 }
 
+void GNADeviceHelper::tagMemoryRegion(void* memPtr, const GNAPluginNS::memory::rRegion tag) {
+    std::unique_lock<std::mutex> lockGnaCalls{ acrossPluginsSync };
+#if GNA_LIB_VER == 2
```

Comment:
Handle GNA1 library single allocation scheme
Also consider to make single allocation using GNA2 library for non-tlv use cases